### ISSUE TITLE
Functional sign-up page for website

### ISSUE THREAD

**Issue by catherineunciano** [IT3A-2016/Paper-Icebergs#31]

**Comment by catherineunciano** [comment]
<?php
	session_start();
	include_once ('db.php');
	
	$error=false;
	
	if(isset($_POST['signup'])){
		$lastname=mysqli_real_escape_string($connection, $_POST['last_name']);
		$firstname=mysqli_real_escape_string($connection, $_POST['first_name']);
		$middlename=mysqli_real_escape_string($connection, $_POST['middle_name']);
		$companyname=mysqli_real_escape_string($connection, $_POST['company_name']);
		$companyaddress=mysqli_real_escape_string($connection, $_POST['address']);
		$phone=mysqli_real_escape_string($connection, $_POST['phone']);
		$ctelnum=mysqli_real_escape_string($connection, $_POST['ctel_num']);
		$departmentname=mysqli_real_escape_string($connection, $_POST['department_name']);
		$dtelnum=mysqli_real_escape_string($connection, $_POST['dtel_num']);
		$email=mysqli_real_escape_string($connection, $_POST['email']);
		$username=mysqli_real_escape_string($connection, $_POST['user_name']);
        $password=mysqli_real_escape_string($connection, $_POST['password']);
		$confirm_password=mysqli_real_escape_string($connection, $_POST['confirm_password']);
		
		$lastname=stripslashes($lastname);
		$firstname=stripslashes($firstname);
		$middlename=stripslashes($middlename);
		$companyname=stripslashes($companyname);
		$companyaddress=stripslashes($companyaddress);
		$phone=stripslashes($phone);
		$ctelnum=stripslashes($ctelnum);
		$departmentname=stripslashes($departmentname);
		$dtelnum=stripslashes($dtelnum);
		$email=stripslashes($email);
		$username=stripslashes($username);
		$password=stripslashes($password);
		$confirm_password=stripslashes($confirm_password);
		
		$trn_date=date("Y-m-d H:i:s");
		
		if(!preg_match("/^[a-zA-Z ]+$/",$lastname)){
			$error=true;
			$lastname_error="Name must contain only alphabets and space.";
		}
		if(!preg_match("/^[a-zA-Z ]+$/",$firstname)){
			$error=true;
			$firstname_error="Name must contain only alphabets and space.";
		}
		if(!preg_match("/^[a-zA-Z ]+$/",$middlename)){
			$error=true;
			$middlename_error="Name must contain only alphabets and space.";
		}
		if(strlen($phone)<11){
				$error=true;
				$phone_error="Phone number must be minimum of 11 integers. Try entering the phone number like 09XXXXXXXXX";
			}
		if(strlen($ctelnum)<10){
				$error=true;
				$ctelnum_error="Telephone number must be minimum of 10 integers. Try entering the telephone number like 541XXXXXXX";
			}
		if(strlen($dtelnum)<10){
				$error=true;
				$dtelnum_error="Telephone number must be minimum of 10 integers. Try entering the telephone number like 541XXXXXXX";
			}
		if(!filter_var($email,FILTER_VALIDATE_EMAIL)){
			$error=true;
			$email_error="Please enter a valid email address.";
		}
		if(strlen($password)<6){
			$error=true;
			$password_error="Password must be minimum of 6 characters.";
		}
		if($password!=$confirm_password){
			$error=true;
			$confirm_password_error="Passwords do not match.";
		}
		if(!$error){
			$sql ="INSERT INTO user (last_name, first_name, middle_name, company_name, department_name, email, user_name, password,  trn_date) VALUES ('".	$lastname."', '".$firstname."', '".$middlename."', '".$companyname."', '".$departmentname."', '".$email."', '".$username."', '".$password."', '".$trn_date."');";
			
			$sql .="INSERT INTO company (company_name, address, phone, ctel_num ) VALUES ('".$companyname."', '".$companyaddress."', '".$phone."', '".$ctelnum."');";
			
			$sql .="INSERT INTO departments (department_name, dtel_num) VALUES ('".$departmentname."', '".$dtelnum."')";
			
			if(mysqli_multi_query($connection, $sql)){
				header('location: signup.php?register=success');
				$successmsg="Successfully registered! <a href='login.php'>Click here to login</a>";
			}
			else{
				$note="Email already exists.";
			}
		}
	}
?>

<!DOCTYPE html>
<html>

<head>
	<title>QWS</title>
	
	<meta charset="utf-8">
	<meta http-equiv="X-UA-Compatible" content="IE=edge">
	<meta name="viewport" content="width=device-width, initial-scale=1, description" content="">
	<meta name="author" content="">
	
	<link rel="stylesheet" href="../css/bootstrap.min.css" type="text/css">
	<link rel="stylesheet" href="http://maxcdn.bootstrapcdn.com/bootstrap/3.3.6/css/bootstrap.min.css">
	
	<script src="../js/smoothscroll.js"></script>
	<script src="../js/webscript.js"></script>
	<script src="../js/jquery.js"></script>
	<script src="../js/bootstrap.min.js"></script>
	<script src="https://ajax.googleapis.com/ajax/libs/jquery/1.12.0/jquery.min.js"></script>
	<script src="http://maxcdn.bootstrapcdn.com/bootstrap/3.3.6/js/bootstrap.min.js"></script>
	
	<style>
	body {background-color:#2E86C1;}
	h1   {color: blue;}
	p    {color: red;}
	</style>
</head>
<body class="cusbodycolor" oncontextmenu="return false">
	<div class="row">
		<div class="col-md-12">
			<div class="col-md-12 well">
				<span style="font-family:'Times New Roman', Times, serif;font-size:20px;"/>Automated Queueing Web-based System
			</div>
		</div>
	</div>
	
	<div class="row" style="padding:20px 0 20px 0;">
		<br><br>
	</div>
	
	<div class="container">
		<div class="row">
			<div class="col-md-12">
				<form role="form" action="<?php echo $_SERVER['PHP_SELF']; ?>" method="post" name="signupform">
					<fieldset>
						<div class="col-md-12 well">
							<div class="row"><div class="col-sm-12 col-md-offset-5">
								<h4><span style="font-family:'Times New Roman', Times, serif;font-size:30px;"/>Register<h4></div></div>
							<div class="col-sm-12">
								<h5><i>USER'S DETAILS</i></h5>
								<div class="col-md-4">
									<small>Last Name</small>
									<input type="text" name="last_name" id="last_name" placeholder="Enter Last Name" class="form-control borderless" style="margin-bottom:10px;" required />
									<span class="text-danger"><?php if (isset($lastname_error)) echo $lastname_error; ?></span>
								</div>
								
								<div class="col-md-4">
									<small>First Name</small>
									<input type="text" name="first_name" id="first_name" placeholder="Enter First Name" class="form-control borderless" style="margin-bottom:10px;" required />
									<span class="text-danger"><?php if (isset($firstname_error)) echo $firstname_error; ?></span>
								</div>
								
								<div class="col-md-4">
									<small>Middle Name</small>
									<input type="text" name="middle_name" id="middle_name" placeholder="Enter Middle Name" class="form-control borderless" style="margin-bottom:10px;" required />
									<span class="text-danger"><?php if (isset($middlename_error)) echo $middlename_error; ?></span>
								</div>
							</div>
							
							<div class="col-sm-12"><hr></div>
							
							<div class="col-sm-12">
								<h5><i>COMPANY DETAILS</i></h5>
								<div class="col-md-6">
									<small>Company Name</small>
									<input type="text" name="company_name" id="company_name" placeholder="Enter Company Name" class="form-control borderless" style="margin-bottom:10px;" required />
								</div>
								<div class="col-md-6">
									<small>Company Address</small>
									<input type="text" name="address" id="address" placeholder="Enter Company Address" class="form-control borderless" style="margin-bottom:10px;" required />
								</div>
								<div class="col-md-6">
									<small>Phone Number</small>
									<input type="number" name="phone" id="phone" placeholder="Enter Phone Number" class="form-control borderless" style="margin-bottom:10px;"/>
									<span class="text-danger"><?php if (isset($phone_error)) echo $phone_error; ?></span>
								</div>
								<div class="col-md-6">
									<small>Telephone Number</small>
									<input type="number" name="ctel_num" id="ctel_num" placeholder="Enter Telephone Number" class="form-control borderless" style="margin-bottom:10px;"/>
									<span class="text-danger"><?php if (isset($ctelnum_error)) echo $ctelnum_error; ?></span>
								</div>
							</div>
							
							<div class="col-sm-12"><hr></div>
							
							<div class="col-sm-12">
								<h5><i>DEPARTMENT DETAILS</i></h5>
								<div class="col-md-6">
									<small>Department Name</small>
									<input type="text" name="department_name" id="department_name" placeholder="Enter Department Name" class="form-control borderless" style="margin-bottom:10px;" required />
								</div>
								<div class="col-md-6">
									<small>Telephone Number</small>
									<input type="number" name="dtel_num" id="dtel_num" placeholder="Enter Telephone Number" class="form-control borderless" style="margin-bottom:10px;"/>
									<span class="text-danger"><?php if (isset($dtelnum_error)) echo $dtelnum_error; ?></span>
								</div>
							</div>
							
							<div class="col-sm-12"><hr></div>
							
							<div class="col-sm-12">
								<h5><i>USER'S ACCOUNT DETAILS</i></h5>
								<div class="col-md-6">
									<small>Email</small>
									<input type="text" name="email" id="email" placeholder="Enter Email Address" class="form-control borderless" style="margin-bottom:10px;" required />
									<span class="text-danger"><?php if (isset($email_error)) echo $email_error; ?></span>
								</div>
								
								<div class="col-md-6">
									<small>Username</small>
									<input type="text" name="user_name" id="user_name" placeholder="Enter Username" class="form-control borderless" style="margin-bottom:10px;" required />
								</div>

								<div class="col-md-6">
									<small>Password</small>
									<input type="password" name="password" id="password" placeholder="Enter Password" class="form-control borderless" style="margin-bottom:10px;" required />
									<span class="text-danger"><?php if (isset($password_error)) echo $password_error; ?></span>
								</div>

								<div class="col-md-6">
									<small>Confirm Password</small>
									<input type="password" name="confirm_password" id="confirm_password" placeholder="Confirm Password" class="form-control borderless" style="margin-bottom:10px;" required />
									<span class="text-danger"><?php if (isset($confirmpassword_error)) echo $confirmpassword_error; ?></span>
								</div>
							</div>
							
							<div class="col-sm-12"><hr></div>
							
							<div class="col-md-4 col-md-offset-10" style="align:right;">
								<input type="submit" name="signup" value="Create Account" class="btn btn-primary"/>
							</div>
						</div>
					</fieldset>
				</form>
			</div>	
		</div>
		
		<div class="row"><span class="note"><?php if (isset($successmsg)) { echo $successmsg; } ?></span></div>
		
		<div class="row">
			<div class="col-md-4 col-md-offset-4 text-center">	
			<span>Already Registered? Click <a href="../login.php" style="color:black">here</a>. </span> 
			</div>
		</div>
	</div>
	
	<div class="row" style="padding:20px 0 20px 0;"><br><br></div>
	<footer style="text-align:center;background-color:#707B7C;color:#000000;">Paper-Icebergs 2017 &copy; All rights reserved.</footer>
</body>
</html>

**Comment by catherineunciano** [comment]
![signup2.jpg](https://zube.io/files/IT3A-2016/92ba66c59bfaa0dc1d65d373d3058170.jpeg)

Guaranteed that the multi-query is working and all the data are pushed through in the database.
